[10.0][IMP] web_timeline improvements

Creamaster opened the pull request:
* Update the vis.js framework to latest stable version (v4.20.1) to add new functionalities and save almost
400ko of source code in odoo assets (the vis dev team has isolated the code for timeline widget from the rest of their widgets)

* Add the possibility to display events with no duration (instantaneous) as the same time has regular events, exemple:
![image](https://user-images.githubusercontent.com/13518832/30909074-19194328-a380-11e7-9ad1-73d691423864.png)


* Add the possibility to choose between two scroll mode :
  -  zoom in time resolution (default)
  or
  -  vertical scroll (web browser default behaviour), exemple on heavy clustured view:
![scroll_vertical_exemple](https://user-images.githubusercontent.com/13518832/30909039-fb1db80e-a37f-11e7-8041-a93452a8c4c5.gif)

Creamaster commented:
In the code, I stumbled on a variable 'all_day' the seems to never be initialize and consequently the test based on this variable could never be triggered, is this an old feature ? Sould this code be updated ?

web_timeline/static/src/js/web_timeline.js line:200
```javascript
all_day = this.all_day ? evt[this.all_day] : false,
```

Creamaster commented:
Also, I parsed the javascript file with my linter, I could revert all the non important changes caused by the reformat, If needed

lmignon commented on web_timeline/static/lib/vis/vis-timeline-graph2d.min.js:
```diff
@@ -0,0 +1,40 @@
+/**
```
@Creamaster PLZ We must keep the minified version of the JS and CSS files to ease the debug. All the js and css files are automatically minified by Odoo when the application run in normal mode. But once you activate the debug mode, you get the original file and it's very useful to ease the debug.

Creamaster replied:
That was my intent, but the non-minified version isn't distributed anymore : 
https://github.com/almende/vis/tree/master/dist

lmignon replied:
@Creamaster Ok

lmignon commented on web_timeline/static/src/js/web_timeline.js:
```diff
@@ -62,12 +62,17 @@ odoo.define('web_timeline.TimelineView', function (require) {
-        parse_colors: function(){
-            if(this.fields_view.arch.attrs.colors) {
-                this.colors = _(this.fields_view.arch.attrs.colors.split(';')).chain().compact().map(function(color_pair) {
+        parse_colors: function () {
```
@Creamaster PLZ you must avoid to change the formatting of the file even if this one is not conform to your criteria. It's hard to dissociate cosmetic changes from the changes required by your PR. These cosmetic changes should at least be done into a separate commit.

pedrobaeza replied:
I think this is for complying with JS linter, which is correct, but as said, better to keep it in a separate commit.

Creamaster replied:
As requested, done in separate commit

lmignon reviewed (changes requested):
@Creamaster Thank you for this enhancement. Comments done into the code

Creamaster committed:
[10.0][IMP] web_timeline improvements:

* Update the vis.js framework to latest stable version (v4.20.1) to add new functionality and save almost
400ko of source code in odoo assets (the vis dev team has isolate the code for timeline widget from the
 rest of their widgets)

* Add the possibility to display events with no duration (instantaneous) as the same time has regular events

* Add the possibility to choose between two scroll mode :
  -  zoom in time resolution (default)
  or
  -  vertical scroll (web browser default behaviour)

Signed-off-by: adrien.didenot <adrien.didenot@horanet.com>

Creamaster committed:
Cosmetic changes (to skip or squash)

Signed-off-by: adrien.didenot <adrien.didenot@horanet.com>

pedrobaeza commented:
@Creamaster thanks for the welcome improvement. Can you check https://github.com/OCA/project/issues/317 to see if you can fix it at the same time with the improvement?

lmignon reviewed (approved):
Thank you for the split of your commits!
:+1:  Code Review

Creamaster commented:
@pedrobaeza I will update the code to fix the [issue 317](https://github.com/OCA/project/issues/317)

pedrobaeza commented:
Thank you, @Creamaster 

https://github.com/OCA/project/issues/316 is also related and give you some hints I think.

Creamaster committed:
[Fix OCA/project#317] Add the property date_delay to TimeLine views (same as base Calendar view)

In Odoo calendar view, the attribute date_delay is an alternative to date_stop, to provides the duration of the event instead of its end date.
This commit integrate this attribute to the Timeline view, but contrary to the base Calendar view the Timeline view could use both arguments (date_stop and date_delay) simultaneously.

Signed-off-by: adrien.didenot <adrien.didenot@horanet.com>

Creamaster committed:
[IMP] Add an option to initialize the timeline window on display.

The default window display all the events (aka 'fit'), in case of events spread over the year, the events are invisibles (too small to be readable)

Signed-off-by: adrien.didenot <adrien.didenot@horanet.com>

Creamaster commented:
A (long) gif to demonstrate the timeline with the added 'date_delay' attribute

I added two datetime fields on the defaut project.task model, and changed the views as follow :

| Calendar view  | Timeline view |
| ------------- | ------------- |
| date_start="x_date_start" date_delay="planned_hours"  | date_start="x_date_start" date_stop="x_date_stop" date_delay="planned_hours" |


![demo_timeline](https://user-images.githubusercontent.com/13518832/30976745-dc9e6abc-a475-11e7-8b80-58ef140557d4.gif)

pedrobaeza reviewed (approved):
You have made an excellent work, thanks.

davinci41 commented:
I noticed a mistake. How to set the start time and duration of the task, and in the calendar and in the timeline view is displayed well. In the timeline you can edit start time and duration. The calendar is displayed correctly. As in the calendar, I will move the task to the next day in the timeline so it displays:
![image](https://user-images.githubusercontent.com/25403547/32292140-6becc5ae-bf3f-11e7-8505-4ebf15e2dd5e.png)
Editing in the calendar, the program does not change the end date
